lambda/internal-release-handler/main.go

```
@@ -0,0 +1,201 @@
+			break
+		}
+
+		opts.Page = resp.NextPage
+	}
+
+	return allIssues, nil
+}
+
+// createIssue creates an issue on the project board and sets it to the Provider Upgrades column
+func createIssue(repoName string, event releaseEvent, ctx context.Context, gitHubClient *github.Client) error {
+	issueTitle := fmt.Sprintf("Upgrade %s to %s", event.Repo, event.TagName)
+
+	log.Printf("Checking for an existing issue in repo '%s'", repoName)
+	issues, err := getIssues(ctx, gitHubClient, repoName)
+	if err != nil {
+		return err
+	}
+
+	issueExists := false
+	for _, issue := range issues {
+		if *issue.Title == issueTitle {
+			issueExists = true
+			break
+		}
+	}
+
+	if issueExists {
+		log.Printf("There is already an issue with the title '%s' in repo 'pulumi/%s'.", issueTitle, repoName)
+		return nil
+	} else {
+		log.Print("Did not find an existing issue.  Creating a new issue.")
+
+		body := fmt.Sprintf("Release details: %s", event.Url)
+
+		issue, _, err := gitHubClient.Issues.Create(ctx, "pulumi", repoName, &github.IssueRequest{
+			Title:  github.String(issueTitle),
+			Labels: &[]string{"kind/enhancement"},
+			Body:   github.String(body),
+		})
+		if err != nil {
+			return err
+		}
+
+		log.Printf("Adding issue to project board.")
+		//platformIntegrationsBoardId := 12058265
+		providerUpgradesColumnsId := int64(14558007)
+		_, _, err = gitHubClient.Projects.CreateProjectCard(ctx, providerUpgradesColumnsId, &github.ProjectCardOptions{
+			ContentID: *issue.ID,
+			// Not documented in the API.  See instead: https://stackoverflow.com/questions/57024087/github-api-how-to-move-an-issue-to-a-project
+			ContentType: "Issue",
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getDownstreamRepos(upstreamRepoName string) []string {
+	switch upstreamRepoName {
+	case "fake-aws":
+		return []string{"fake-awsx"}
+	case "pulumi-aws":
+		return []string{"pulumi-awsx", "pulumi-eks", "pulumi-aws-apigateway"}
+	default:
+		return []string{}
```

would it make sense to have this return an error also so you could return an helpful error in the default case, or perhaps log a message?

---

I agree - that's why the caller logs a message!

https://github.com/pulumi/github-issue-automation/pull/31/files#diff-bbbe637f3c28eb20497cc5df51b93b397e194707854eb656d06c1a01432a325aR52